src/analysisd/mitre.c

```diff
@@ -0,0 +1,172 @@
+    cJSON *tactic;
+
+    snprintf(path_db, sizeof(path_db), "%s/%s.db", WDB_DIR, WDB_MITRE_NAME);
+
+    /* Create hash table */
+    mitre_table = OSHash_Create();
+    
+    /* Get Mitre IDs from Mitre's database */
+    os_calloc(OS_SIZE_6144 + 1, sizeof(char), wazuhdb_query);
+    snprintf(wazuhdb_query, OS_SIZE_6144, "mitre sql SELECT id from attack;");
+    if (result = wdb_send_query(wazuhdb_query, &response), result == -2) {
+        mdebug1("Mitre info loading failed. Unable to connect to socket '%s'.", WDB_LOCAL_SOCK);
+        merror("Mitre matrix information could not be loaded.");
+        goto end;
+    }
+
+    if (result == -1) {
+        mdebug1("Mitre info loading failed. No response or bad response from wazuh-db: %s", response);
+        merror("Mitre matrix information could not be loaded.");
+        os_free(response);
+        goto end;
+    }
+
+    /* Parse IDs string */
+    if (root = cJSON_Parse(response+3), !root) {
+        mdebug1("Mitre info loading failed. Query response cannot be parsered: %s", response);
+        merror("Mitre matrix information could not be loaded.");
+        os_free(response);
+        cJSON_Delete(root);
+        result = -1;
+        goto end;
+    }
+
+    /* Response parameter has to be freed before continuing*/
+    os_free(response);
+
+    /* Getting array size */
+    if (size_ids = cJSON_GetArraySize(root), size_ids == 0) {
+        mdebug1("Mitre info loading failed. Query response has 0 elements.");
+        merror("Mitre matrix information could not be loaded.");
+        cJSON_Delete(root);
+        result = -1;
+        goto end;
+    }
+    
+    for (i=0; i<size_ids; i++){
+        /* Getting Mitre attack ID from Mitre's database in Wazuh-DB  */
+        ids = cJSON_GetArrayItem(root, i);
+        if (id = cJSON_GetObjectItem(ids,"id"), id == NULL) {
+            mdebug1("Mitre info loading failed. It was not possible to get cJSON item from IDs array.");
+            merror("Mitre matrix information could not be loaded.");
+            cJSON_Delete(root);
+            result = -1;
+            goto end; 
+        }
+        ext_id = id->valuestring;
+
+        /* Consulting Mitre's database to get Tactics */
+        snprintf(wazuhdb_query, OS_SIZE_6144, "mitre sql SELECT phase_name FROM has_phase WHERE attack_id = '%s';", ext_id);
+        if (result = wdb_send_query(wazuhdb_query, &response), result == -2) {
+            mdebug1("Mitre info loading failed. Unable to connect to socket '%s'.", WDB_LOCAL_SOCK);
+            merror("Mitre matrix information could not be loaded.");
+            cJSON_Delete(root);
+            goto end;
+        }
+
+        if (result == -1) {
+            mdebug1("Mitre info loading failed. No response or bad response from wazuh-db: %s", response);
+            merror("Mitre matrix information could not be loaded.");
+            cJSON_Delete(root);
+            os_free(response);
+            result = -1;
+            goto end;
+        }
```

Comment:
Hello @danimegar,

In this case, you enter to the conditional if result value is -1. It's not necessary to re-assign -1 to result variable on line `102`.

Regards.

Reply:
It's done.